# The optimized rounding implementation used by ANGLE doesn't work on every platform,
# in particular 32-bit x86 Windows using MinGW.
# This patch conditionally disables the optimized rounding implementation if we detect at configure time that it's broken.

--- a/src/common/mathutil.h
+++ b/src/common/mathutil.h
@@ -606,7 +606,7 @@ inline R roundToNearest(T input)
 {
     static_assert(std::is_floating_point<T>::value);
     static_assert(std::numeric_limits<R>::is_integer);
-#if defined(__aarch64__) || defined(_M_ARM64)
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(MKXPZ_ANGLE_NO_OPTIMIZED_ROUND)
     // On armv8, this expression is compiled to a dedicated round-to-nearest instruction
     return static_cast<R>(std::round(input));
 #else
